### PR TITLE
Is488/auth ui extension

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?
UI to:
- [x] reset password request and allow
- [x] reset email (logged in)
- [x] reset password (logged in)

image

image

## Related issue number
closes case #448 
connected to #448 

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [x] **Runs in the swarm**
- [x] If you design a new module, add your user to .github/CODEOWNERS

[waffle.io]:https://waffle.io/marketing-assets/documents/waffleio_cheatsheet_v1.pdf?utm_source=blog&utm_medium=cheatsheet-ctabutton&utm_campaign=cheatsheet

### CODE REVIEW

File: services/web/client/source/class/qxapp/auth/core/Utils.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            valid: false
+          });
+        });
+      }
+      return isValid;
+    },
+
+    findGetParameter: function(parameterName) {
+      let result = null;
+      const params = window.location.search.substr(1).split("&");
```
Comment (oetiker) [inline services/web/client/source/class/qxapp/auth/core/Utils.js]: this seems dangerous as a rule, reading data of the url in the browser.  if this is necessary at all, I would place parameters into the part following the # character as this is explicitly thought to be the part responsible for navigating inside the page using anchor tags.

File: services/web/client/source/class/qxapp/desktop/preferences/pages/BasePage.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  extend: qx.ui.tabview.Page,
+
+  construct: function(title, iconSrc = null) {
+    this.base(arguments, null, iconSrc);
+
+    this.setLayout(new qx.ui.layout.VBox(10).set({
+      spacing: 10,
+      alignX: "center"
+    }));
+
```
Comment (oetiker) [inline services/web/client/source/class/qxapp/desktop/preferences/pages/BasePage.js]: having the icons above the label might look better in this context

```javascript
this.getChildControl("button").set({
  iconPosition: "top",
  center: true,
  paddingRight: 20
});
```

File: services/web/client/source/class/qxapp/desktop/preferences/pages/ProfilePage.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+          "firstName": data["first_name"] || "",
+          "lastName": data["last_name"] || "",
+          "email": data["login"],
+          "role": data["role"] || ""
+        });
+      });
+
+      request.addListenerOnce("fail", function(e) {
+        const error = e.getTarget().getResponse().error;
+        const msg = error ? error["errors"][0].message : "Failed to get profile";
```
Comment (oetiker) [inline services/web/client/source/class/qxapp/desktop/preferences/pages/ProfilePage.js]: translation is missing

File: services/web/client/source/class/qxapp/desktop/preferences/pages/SecurityPage.js [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+    __tokensList: null,
+
+    __createTokensSection: function() {
+      // layout
+      let box = this._createSectionBox(this.tr("Access Tokens"));
+
+      let label = this._createHelpLabel(
+        "List of API tokens to access external services. Currently, \
+         only <a href='https://app.blackfynn.io'>DAT-Core</a> API keys are supported."
```
Comment (oetiker) [inline services/web/client/source/class/qxapp/desktop/preferences/pages/SecurityPage.js]: translation is missing

File: services/web/client/source/class/qxapp/auth/core/Utils.js (hunk 1 above)
Comment (pcrespov) [reply services/web/client/source/class/qxapp/auth/core/Utils.js]: Now we use fragment as ``#page=reset-password;code=123456`` so browser is not reloaded and front-end switches to the right page

image

File: services/web/client/source/class/qxapp/desktop/preferences/pages/BasePage.js (hunk 2 above)
Comment (pcrespov) [reply services/web/client/source/class/qxapp/desktop/preferences/pages/BasePage.js]: No text

File: services/web/client/source/class/qxapp/auth/core/Utils.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+qx.Class.define("qxapp.auth.core.Utils", {
+  type: "static",
+
+  statics:
+  {
+    checkPasswordSecure:  function(password, itemForm) {
+      const isValid = password !== null && password.length > 2;
+      if (!isValid) {
+        const msg = qx.locale.Manager.tr("Please enter a password at with least 3 characters.");
+        itemForm.setInvalidMessage(msg);
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply services/web/client/source/class/qxapp/auth/core/Utils.js]: Added a call to webserver and the quality of the password is flashed to the user but is not required

**Pull request merged**